File: advent02/advent2a.py
import os
os.chdir("/home/jannik/Documents/coding_advent/advent2/")
with open("input.txt", "r") as fp:
    intcode = [int(i) for i in fp.readline().split(",")]

print(len(intcode))
print(intcode)
intcode[1] = 12
intcode[2] = 2

pos = 0

while True:
    curr_value = intcode[pos]
    if curr_value == 99:
        break
    if curr_value == 1:
        intcode[intcode[pos+3]] = intcode[intcode[pos+1]] + intcode[intcode[pos+2]]
    if curr_value == 2:
        intcode[intcode[pos+3]] = intcode[intcode[pos+1]] * intcode[intcode[pos+2]]
    pos += 4

print(intcode[0])

File: advent02/advent2a.cc
#include <fstream>
#include <iostream>
#include <string>
using namespace std;

int main() {
  string line;
  int pos {-1}, last_pos, op;
  int intcode[129];

  ifstream advent2("advent2/input.txt");
  if(advent2.is_open()) {
    getline(advent2, line);
    advent2.close();
  }

  for(int& i : intcode) {
    last_pos = pos + 1;
    pos = line.find(",", last_pos);
    i = stoi(line.substr(last_pos, (pos == -1) ? line.size() : pos - last_pos));
  }

  intcode[1] = 12;
  intcode[2] = 2;

  for(int i = 0;; i += 4) {
    op = intcode[i];
    if(op == 99) break;
    if(op == 1) intcode[intcode[i + 3]] = intcode[intcode[i + 1]] + intcode[intcode[i + 2]];
    if(op == 2) intcode[intcode[i + 3]] = intcode[intcode[i + 1]] * intcode[intcode[i + 2]];
  }

  for(int i : intcode) cout << i << " ";
  return 0;
}
